### ISSUE TITLE
Displacement size inference changed between 2.0.0 and 3.0.0

### ISSUE THREAD

**Issue by gabi-250** [CensoredUsername/dynasm-rs#103]
When upgrading [our `dynasmrt` dependency to 3.0.0](https://gitlab.torproject.org/tpo/core/arti/-/merge_requests/2526), we noticed the generated code is slightly larger than before (previously we were using `dynasm 2.0.0`).

As far as I can tell, the difference is that dynasm 3.0.0 generates a 4-byte displacement, whereas 2.0.0 generates a 1-byte one (perhaps something changed in the displacement size calculations?). Minimal repro:
```rust
use dynasmrt::{dynasm, DynasmApi};

fn main() {
    let mut ops = dynasmrt::SimpleAssembler::new();

    dynasm!(ops
        ; .arch x64
        ; lea ecx, [esi - 1]
    );

    let buf = ops.finalize();

    // With dynasm 2.0.0, prints
    // len = 4
    // [67, 8d,            4e,              ff]
    //      ^ lea opcode   ^ ModR/M byte    ^ 8-bit displacement
    //
    // With dynasm 3.0.0, prints
    // len = 7
    // [67, 8d,            8e,              ff, ff, ff, ff]
    //      ^ lea opcode   ^ ModR/M byte    ^ 32-bit displacement
    println!("len = {}", buf.len());
    println!("{:x?}", buf);
}
```

cc @CensoredUsername

**Comment by CensoredUsername** [comment]
Thanks for the test case, Seems like this was only happening with negative displacements, which are quite rare.

It's fixed in 73395f8, and should now be live as part of v3.0.1.

**Comment by CensoredUsername** [comment]
Closing this as it seems to have been fixed.
